docs/build.rb

```diff
@@ -0,0 +1,1151 @@
+    def initialize(config)
+      @config = config
+      @links = {}
+    end
+
+    def extract(markdown)
+      extracted_absolute_wiki_links = extract_absolute_wiki_links(markdown)
+      @links = @links.merge(extracted_absolute_wiki_links)
+
+      extracted_relative_links = extract_relative_links(markdown)
+      @links = @links.merge(extracted_relative_links)
+
+      @links
+    end
+
+    def rerender(markdown)
+      links ||= @links
+
+      new_markdown = markdown.clone
+      links.each_value do |link|
+        new_markdown.gsub!(link[:full_match], link[:replacement])
+      end
+
+      fix_github_username_links(new_markdown)
+    end
+
+    attr_reader :links
+
+    protected
+
+    def pages
+      @config.enum_for(:each).map { |page| page }
+    end
+
+    # scans for absolute links to the old wiki such as 'https://github.com/rapid7/metasploit-framework/wiki/Metasploit-Web-Service'
+    def extract_absolute_wiki_links(markdown)
+      new_links = {}
+
+      markdown.scan(%r{(https?://github.com/rapid7/metasploit-framework/wiki/([\w().%_-]+))}) do |full_match, old_path|
+        full_match = full_match.gsub(/[).]+$/, '')
+        old_path = URI.decode_www_form_component(old_path.gsub(/[).]+$/, ''))
+
+        new_path = new_path_for(old_path)
+        replacement = "{% link docs/#{new_path} %}"
+
+        link = {
+          full_match: full_match,
+          type: :absolute,
+          new_path: new_path,
+          replacement: replacement
+        }
+
+        new_links[full_match] = link
+      end
+
+      new_links
+    end
+
+    # Scans for substrings such as '[[Reference Sites|Git Reference Sites]]'
+    def extract_relative_links(markdown)
+      existing_links = @links
+      new_links = {}
+      markdown.scan(/(\[\[([\w_ '().:,-]+)(?:\|([\w_ '():,.-]+))?\]\])/) do |full_match, left, right|
+        old_path = (right || left)
+        new_path = new_path_for(old_path)
+        raise "Mismatching new_path for #{old_path}" if existing_links[full_match] && existing_links[full_match][:path]
+
+        link_text = left
+        replacement = "[#{link_text}]({% link docs/#{new_path} %})"
+
+        link = {
+          full_match: full_match,
+          type: :relative,
+          left: left,
+          right: right,
+          new_path: new_path,
+          replacement: replacement
+        }
+
+        new_links[full_match] = link
+      end
+
+      new_links
+    end
+
+    def new_path_for(old_path)
+      old_path = old_path.gsub(' ', '-')
+      matched_pages = pages.select do |page|
+        !page[:folder] &&
+          page.fetch(:path).downcase.end_with?(old_path.downcase + '.md')
+      end
+      if matched_pages.empty?
+        raise "Missing path for #{old_path}"
+      end
+      if matched_pages.count > 1
+        raise "Duplicate paths for #{old_path}"
+      end
+
+      matched_pages.first.fetch(:new_path)
+    end
+
+    def fix_github_username_links(content)
+      known_github_names = [
```

Comment:
I have some concerns r.e the maintainability of this list of names in the long run, do we have any way around this?

Reply:
This migration logic will be deleted in the future, after we commit the migrated wiki files into the main repository. i.e. I've no concerns with the maintainability of this script, as it won't exist soon 🤞 

There's a comment to that effect at the top of the module too, but always good to call it out explicitly 📈

Reply:
Sounds good to me, thought this script might remain around for a while which is why I raised this but if its being deleted soon then I'll mark this as resolved.